File: 75-sort-colors/sort-colors.py
from collections import Counter
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        low, mid, high = 0, 0, len(nums) - 1

        while mid <= high:
            if nums[mid] == 0:
                # Swap nums[mid] and nums[low]
                nums[low], nums[mid] = nums[mid], nums[low]
                low += 1
                mid += 1
            elif nums[mid] == 2:
                # Swap nums[mid] and nums[high]
                nums[mid], nums[high] = nums[high], nums[mid]
                high -= 1
            else:
                # nums[mid] == 1, just move mid forward
                mid += 1

File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {

        int low = 0;
        int mid = 0;
        int high = nums.length - 1;

        while (mid <= high) {
            if (nums[mid] == 0) {

                int temp = nums[mid];
                nums[mid] = nums[low];
                nums[low] = temp;

                low += 1;
                mid += 1;
            } else if (nums[mid] == 1) {
                mid += 1;
            } else {

                int temp = nums[mid];
                nums[mid] = nums[high];
                nums[high] = temp;
                
                high -= 1;
            }
        }

        // for (int i = 1; i < nums.length; i++) {
        //     for (int j = 0; j  < i; j ++) {

        //         if (nums[i] < nums[j]) {
        //             int temp = nums[i];
        //             nums[i] = nums[j];
        //             nums[j] = temp;
        //         }
        //     }
        // }
    }
}